packages/adapter-drizzle/src/mysql/index.ts

```diff
@@ -0,0 +1,254 @@
+import { and, eq } from "drizzle-orm"
+import crypto from 'node:crypto'
+import {
+  int,
+  timestamp,
+  mysqlTable,
+  primaryKey,
+  varchar,
+  MySqlTextColumnType
+} from "drizzle-orm/mysql-core"
+import type { Adapter, AdapterAccount } from "@auth/core/adapters"
+import { drizzle } from "drizzle-orm/mysql2"
+import { createPool } from "mysql2/promise"
+
+export const users = mysqlTable("users", {
+  id: varchar("id", { length: 255 }).notNull().primaryKey(),
+  name: varchar("name", { length: 255 }),
+  email: varchar("email", { length: 255 }).notNull(),
+  emailVerified: timestamp("emailVerified", { mode: "date" }),
+  image: varchar("image", { length: 255 }),
+})
+
+export const accounts = mysqlTable(
+  "accounts",
+  {
+    userId: varchar("userId", { length: 255 })
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
```

Comment:
Planet scale does not support foreign keys. So I had to remove `.references` to be able to push the migrations to their servers.

<img width="1290" alt="image" src="https://github.com/nextauthjs/next-auth/assets/49499/0d4da236-b2a8-4b96-9ad3-5edbb5e37847">

Reply:
I'm also using Planetscale but have you seen this `packages/adapter-drizzle/src/planetscale/index.ts`?

Reply:
It's what I'm using. The problem is I think that Planet Scale use a flavor of MySQL called Vitess which is mean to be scalable and they don't allow foreign keys. That's my theory. 

Any way this issue goes away not using `.references`. You don't have foreign keys but that's the price for using something at planet scale (pum intended)

Reply:
There are two seperate adapters, one for mysql and one for planetscale in the PR.

Reply:
Oh you're right my bad. Ignore my comment here